### PR TITLE
client: Convert some jQuery to JS Promises

### PR DESCRIPTION
**Pull request by martinpitt**
Keep cockpit.defer() for run_transaction() as that uses progress
notify.
   
connect() will be ported separately as that is very complex.

### CODE REVIEW

**Comment by martinpitt** [comment]
Meh, I don't understand the [continuous-atomic failure](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-114-20200721-125905-f830a42e-continuous-atomic/log.html#5-2), and I don't feel like tracking this down now. This is really not important, so let's just do that small cleanup for now.

**Comment by martinpitt** [comment]
@KKoukiou : Do you have some time to review this in the next days? Thanks!

File: src/client.js [hunk 1]
```diff
@@ -700,9 +696,11 @@ class RPMOSTreeDBusClient {
                                     }
                                 });
                             transaction_client.call("/", TRANSACTION, "Start");
```
Comment (KKoukiou) [inline src/client.js:698]: Failures from this call won't be caught in the fail unless this is returned.

Comment (martinpitt) [reply src/client.js:698]: Right, the code doesn't change that. Perhaps we should do that though, good catch!

**Pull request merged**